src/icainfo.c

```diff
@@ -38,6 +41,225 @@
 #endif
 #define COPYRIGHT "Copyright IBM Corp. 2007, 2021."
 
+#define CELL_SIZE 3
+
+typedef struct {
+	unsigned int nid;
+	unsigned char *sname;
+	unsigned int flags;
+} s390_supported_curves_t;
+
+static unsigned int cca_nids[] = {
+	NID_X9_62_prime192v1, NID_secp224r1, NID_X9_62_prime256v1,
+	NID_secp384r1, NID_secp521r1, NID_brainpoolP160r1, NID_brainpoolP192r1,
+	NID_brainpoolP224r1, NID_brainpoolP256r1, NID_brainpoolP320r1,
+	NID_brainpoolP384r1, NID_brainpoolP512r1 };
+static unsigned int cca_nids_len = sizeof(cca_nids) / sizeof(unsigned int);
+
+static unsigned int cpacf_nids[] = {
+	NID_X9_62_prime256v1, NID_secp384r1, NID_secp521r1,
+	NID_ED25519, NID_ED448, NID_X25519, NID_X448 };
+static unsigned int cpacf_nids_len = sizeof(cpacf_nids) / sizeof(unsigned int);
+
+static int is_msa9(void)
```

Comment:
Shouldn't this be some form of `stfle` or checking the facility bits in `/proc/cpuinfo` for facility 155?

Reply:
Hm, ... libica itself uses STFLE to determine the hardware capabilities and sets up the function list accordingly. I wanted to avoid this overhead in the icainfo utility and just check the functionlist instead. The idea is then: if ECDSA_SIGN is supported via static hw (i.e. CPACF), then we are at MSA9.

Reply:
Okay, if we have CPACF available (i.e., not a NO_CPACF build), then this might work.